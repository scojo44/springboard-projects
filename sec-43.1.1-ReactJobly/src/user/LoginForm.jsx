import React, { useState } from 'react'
import {useForm} from 'react-hook-form'
import {useNavigate} from 'react-router-dom'
import JoblyApi from '../api'
import Alert from '../widgets/Alert'
// import './LoginForm.css'

export default function LoginForm() {
  const {register, handleSubmit, formState: {errors}} = useForm();
  const [loginError, setLoginError] = useState();
  const navigate = useNavigate();

  return (
    <form className="LoginForm" onSubmit={handleSubmit(loginUser)}>
      <h2>Log In</h2>

      <p>
        <label htmlFor="username">Username: </label>
        <input {...register("username", {required: true})} />
        {errors.usernameRequired && <span className="input-error">Please enter your username</span>}
      </p>
      <p>
        <label htmlFor="password">Password: </label>
        <input type="password" {...register("password", {required: true})} />
        {errors.passwordRequired && <span className="input-error">Please enter your password</span>}
      </p>

      <button type="submit">Log In</button>

      {loginError && <Alert type="error" messages={[`Error logging in: ${loginError}`]} />}
    </form>
  )

  async function loginUser({username, password}) {
    try {
      const token = await JoblyApi.login(username, password);
      setLoginError(null);
      navigate('/');
    }
    catch(e) {
      setLoginError(e);
    }
  }
}
