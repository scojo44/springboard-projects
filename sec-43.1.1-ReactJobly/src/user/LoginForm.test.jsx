import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import LoginForm from './LoginForm'

describe('LoginForm Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <LoginForm />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <LoginForm />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
