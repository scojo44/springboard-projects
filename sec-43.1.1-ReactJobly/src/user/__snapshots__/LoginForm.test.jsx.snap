// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginForm Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="LoginForm"
  >
    <h2>
      Log In
    </h2>
    <p>
      <label
        for="username"
      >
        Username: 
      </label>
      <input
        name="username"
      />
    </p>
    <p>
      <label
        for="password"
      >
        Password: 
      </label>
      <input
        name="password"
        type="password"
      />
    </p>
    <button
      type="submit"
    >
      Log In
    </button>
  </form>
</DocumentFragment>
`;
